packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -21,15 +21,30 @@ const
  *
  * @typedef {Object} gridListItemSizeShape
  * @memberof ui/VirtualList
- * @property {Number}    minWidth    The minimum width of the grid list item.
- * @property {Number}    minHeight    The minimum height of the grid list item.
+ * @property {Number} minWidth The minimum width of the grid list item.
+ * @property {Number} minHeight The minimum height of the grid list item.
  * @public
  */
 const gridListItemSizeShape = PropTypes.shape({
 	minHeight: PropTypes.number.isRequired,
 	minWidth: PropTypes.number.isRequired
 });
 
+/**
+ * The shape for the list different item size
+ * in a list for [VirtualList]{@link ui/VirtualList.VirtualList}.
+ *
+ * @typedef {Object} listDifferentItemSizeShape
+ * @memberof ui/VirtualList
+ * @property {Number} minWidth The minimum width of the grid list item.
+ * @property {Number[]} minHeight The minimum height of the grid list item.
+ * @public
+ */
+const listDifferentItemSizeShape = PropTypes.shape({
+	minSize: PropTypes.number.isRequired,
+	size: PropTypes.arrayOf(PropTypes.number).isRequired
```

Comment:
Could you please check this? It probably is not a 'required' one.

Reply:
Good catch! When any item size is not known, then app developers could skip to pass `size` property in `itemSize` prop.